src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -156,19 +174,47 @@ const SignOrExecuteForm = ({
     // If no txId was provided, it's an immediate execution of a new tx
     const id = txId || (await proposeTx(createdTx))
     const txOptions = getTxOptions(advancedParams, currentChain)
+    trackEvent({ ...MODALS_EVENTS.PROPOSE_TX, label: ExecutionType.CONNECTED_WALLET })
     await dispatchTxExecution(createdTx, txOptions, id, onboard, safe.chainId)
 
     return id
   }
 
+  const onRelay = async () => {
+    const [, createdTx, onboard] = assertDependencies()
+
+    const { gasLimit } = getTxOptions(advancedParams, currentChain)
+
+    let id = txId
+    let safeTx = createdTx
+    // Add missing signature
+    if (!hasEnoughSignatures(createdTx, safe)) {
+      const signedTransaction = await dispatchTxSigning(createdTx, safe.version, onboard, safe.chainId, txId)
+
+      if (!signedTransaction) {
+        throw new Error('Could not sign transaction')
+      }
+
+      id = await proposeTx(signedTransaction)
+      safeTx = signedTransaction
+    }
+
+    if (!id) {
+      throw new Error('Transaction could not be proposed')
+    }
+
+    trackEvent({ ...MODALS_EVENTS.PROPOSE_TX, label: ExecutionType.RELAYER })
```

Comment:
Tracking right in the modal is no bueno. We have a tx tracking hook based on tx events.